signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/aws/CachedAwsKmsClientFactory.java

```diff
@@ -13,25 +13,34 @@
 package tech.pegasys.web3signer.signing.secp256k1.aws;
 
 import java.net.URI;
+import java.util.Map;
 import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
 
 import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
 import software.amazon.awssdk.regions.Region;
 import software.amazon.awssdk.services.kms.KmsClient;
 import software.amazon.awssdk.services.kms.KmsClientBuilder;
 
-/** Factory class that provide instance of KmsClient */
-public class AwsKmsClientFactory {
+/** Factory class that provide cached instance of KmsClient */
+public class CachedAwsKmsClientFactory {
+  private static final Map<AwsKmsClientKey, AwsKmsClient> CACHE = new ConcurrentHashMap<>();
```

Comment:
I guess it doesn't make sense to ever clear this cache - object should be tied to the app lifecycle anyway?